src/curve.ts

```diff
@@ -226,11 +229,19 @@ class Curve implements ICurve {
             providerSettings = providerSettings as { network?: Networkish, apiKey?: string };
             this.provider = new ethers.AlchemyProvider(providerSettings.network, providerSettings.apiKey);
             this.signer = null;
+        } else if (providerType.toLowerCase() === 'NoRPC'.toLowerCase()) {
+            providerSettings = providerSettings as { chainId: number, networkName: string };
+            this.isNoRPC = true;
+            this.abstractProvider = {
+                chainId: providerSettings.chainId as number,
+                name: providerSettings.networkName as string,
+            }
```

Comment:
This abstract provider isn't a provider and isn't abstract. Why don't we just keep no provider?

```suggestion
            this.provider = null
```